core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -95,10 +98,17 @@
      */
     @BeforeClass
     public static void init() throws IOException {
+        Version version = Version.CURRENT;
+        // sometimes we need pre-1.3 index version to simulate disabled FieldNamesFieldMapper (e.g. for ExistQueryBuilder)
+        if (randomIntBetween(0, 5) == 0) {
+            version = Version.V_1_2_0;
+        }
```

Comment:
it would uncover other paths I guess, if not now later on. I would love to try and stay as generic as possible rather than having a lot of different situations like this one needing comments to explain why (e.g. xyz query needs it because...), especially with a specific version. hopefully this wouldn't complicate tests more, if it does then I agree with you.